Dockerfile

```diff
@@ -13,7 +13,7 @@
 # limitations under the License.
 
 
-FROM python:3-slim-buster
+FROM python:3.8-slim-buster
```

Comment:
According to https://hub.docker.com/_/python, `3-slim-buster` corresponds to python 3.11 but now we are downgrading to 3.8, why?

Reply:
it corresponds to `python3.7` from what I know. So I upgraded to `Python3.8`

Reply:
If you check the page I linked you can see that these are all the same images: 3.11.2-slim-buster, 3.11-slim-buster, 3-slim-buster, slim-buster

Reply:
Can you instead try adding a `RUN apt-get install -y g++` to the dockerfile to see if that solves it?

Reply:
I am afraid, it does output the same error:

```
...
#12 6.048 Collecting grpcio<2.0dev,>=1.33.2
#12 6.086   Downloading grpcio-1.51.3.tar.gz (22.1 MB)
#12 8.259      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 22.1/22.1 MB 10.1 MB/s eta 0:00:00
#12 9.315   Preparing metadata (setup.py): started
#12 9.628   Preparing metadata (setup.py): finished with status 'error'
#12 9.635   error: subprocess-exited-with-error
#12 9.635   
#12 9.635   × python setup.py egg_info did not run successfully.
#12 9.635   │ exit code: 1
#12 9.635   ╰─> [14 lines of output]
#12 9.635       Traceback (most recent call last):
#12 9.635         File "<string>", line 2, in <module>
#12 9.635         File "<pip-setuptools-caller>", line 34, in <module>
#12 9.635         File "/tmp/pip-install-99ma6hvk/grpcio_0a81b05cdcf9441c808fc788f2c54d42/setup.py", line 262, in <module>
#12 9.635           if check_linker_need_libatomic():
#12 9.635              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
#12 9.635         File "/tmp/pip-install-99ma6hvk/grpcio_0a81b05cdcf9441c808fc788f2c54d42/setup.py", line 209, in check_linker_need_libatomic
#12 9.635           cpp_test = subprocess.Popen(cxx + ['-x', 'c++', '-std=c++14', '-'],
#12 9.635                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
#12 9.635         File "/usr/local/lib/python3.11/subprocess.py", line 1024, in __init__
#12 9.635           self._execute_child(args, executable, preexec_fn, close_fds,
#12 9.635         File "/usr/local/lib/python3.11/subprocess.py", line 1901, in _execute_child
#12 9.635           raise child_exception_type(errno_num, err_msg, err_filename)
#12 9.635       FileNotFoundError: [Errno 2] No such file or directory: 'c++'
#12 9.635       [end of output]
#12 9.635   
#12 9.635   note: This error originates from a subprocess, and is likely not a problem with pip.
#12 9.637 error: metadata-generation-failed
#12 9.637 
#12 9.637 × Encountered error while generating package metadata.
#12 9.637 ╰─> See above for output.
#12 9.637 
#12 9.637 note: This is an issue with the package mentioned above, not pip.
#12 9.637 hint: See above for details.
#12 9.653 
#12 9.653 [notice] A new release of pip available: 22.3.1 -> 23.0.1
#12 9.653 [notice] To update, run: pip install --upgrade pip
------
executor failed running [/bin/sh -c pip install .]: exit code: 1
```

using this

```

FROM python:3-slim-buster

RUN mkdir /home/sa_scanner
COPY src/ /home/sa_scanner/
COPY pyproject.toml /home/sa_scanner/
COPY README.md /home/sa_scanner

WORKDIR /home/sa_scanner
RUN apt-get install -y g++

RUN pip install .


ENTRYPOINT ["gcp-scanner"]
```

Reply:
What about `RUN apt-get install -y build-essential` instead? I'm really not a fan of locking down to a fairly old Python version as we want to test that everything works on up-to-date Python versions.

Reply:
I tried it, which led to an error so I inserted an `RUN apt-get update` to make it work. It ended in an endless wheel
```
 => [7/9] RUN apt-get update                                                                                                                      2.7s
 => [8/9] RUN apt-get install -y build-essential                                                                                                 11.9s
 => [9/9] RUN pip install .                                                                                                                     130.7s 
 => => #   Building wheel for pyu2f (setup.py): finished with status 'done'                                                                            
 => => #   Created wheel for pyu2f: filename=pyu2f-0.1.5-py3-none-any.whl size=39401 sha256=1dbc3d3135c35f20bcf4ccfdd4c0823ec54aae36ce2b298782f28265ce 
 => => # c3a2d3                                                                                                                                        
 => => #   Stored in directory: /root/.cache/pip/wheels/a0/84/f9/1676c313d1bf6c4c30f783a7c5b18648622d5f1b2f4caf468a                                    
 => => #   Building wheel for grpcio (setup.py): started                                                                                               
 => => #   Building wheel for grpcio (setup.py): still running...   
 ```

I also tried 
```
pip install --upgrade pip
pip install --upgrade setuptools wheel
```
which also didn't work.

I suggest using `python:3.10-bullseye`. It works and the version wouldn't be so old. What do you think?